packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts

```diff
@@ -511,10 +482,27 @@ export default ({
     yield put(A.initAmountForm())
   }
 
+  const handleSwapMaxAmountClick = function* ({
+    payload
+  }: ReturnType<typeof A.handleSwapMaxAmountClick>) {
+    const { amount } = payload
+
+    yield put(actions.form.change('swapAmount', 'amount', amount))
+  }
+
+  const handleSwapMinAmountClick = function* ({
+    payload
+  }: ReturnType<typeof A.handleSwapMinAmountClick>) {
+    const { amount } = payload
+
+    yield put(actions.form.change('swapAmount', 'amount', amount))
+  }
```

Comment:
these two functions are exactly the same. do we need both?

Reply:
Because then we can analyse if it was a max or a min click, meaning that it's easier to see on the redux events and easier to track.